source/Cosmos.System2/Graphics/Canvas.cs

```diff
@@ -855,7 +855,48 @@ public virtual void DrawImage(Image image, int x, int y)
                 }
             }
         }
-
+        
+        /// <summary>
+        /// Draw a Scaled Bitmap.
+        /// </summary>
+        /// <param name="image">Image to Scale.</param>
+        /// <param name="x">X coordinate.</param>
+        /// <param name="y">Y coordinate.</param>
+        /// <param name="w">Desired Width.</param>
+        /// <param name="h">Desired Height.</param>
+        private int[] scaleImage(Image image, int newWidth, int newHeight)
+        {
+            int[] pixels = image.rawData;
+            int w1 = (int)image.Width;
+            int h1 = (int)image.Height;
+            int[] temp = new int[newWidth * newHeight];
+            int x_ratio = (int)((w1 << 16) / newWidth) + 1;
+            int y_ratio = (int)((h1 << 16) / newHeight) + 1;
+            int x2, y2;
+            for (int i = 0; i < newHeight; i++)
+            {
+                for (int j = 0; j < newWidth; j++)
+                {
+                    x2 = ((j * x_ratio) >> 16);
+                    y2 = ((i * y_ratio) >> 16);
+                    temp[(i * newWidth) + j] = pixels[(y2 * w1) + x2];
+                }
+            }
+            return temp;
+        }
+        public virtual void DrawScaledImage(Image image, int x, int y,int w,int h)
```

Comment:
Can you please change this method signature to `public virtual void DrawImage(Image image, int x, int y, int width, int height)` ?